### PR TITLE
fix: `array_slice` and `array_element` panicked on empty args

### LINKED ISSUE #10791: `array_slice` panicked when called with empty args
### Describe the bug

thread 'main' panicked at datafusion/functions-array/src/extract.rs:255:39:
index out of bounds: the len is 0 but the index is 0

### To Reproduce

Run the query in CLI
```sh
DataFusion CLI v38.0.0
> select array_slice();

thread 'main' panicked at datafusion/functions-array/src/extract.rs:255:39:
index out of bounds: the len is 0 but the index is 0
```

### Expected behavior

Return an error instead of panic.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?

Closes #10791.

## Rationale for this change
When the `args` are empty, they panic at [display_name](https://github.com/apache/datafusion/blob/ece7ae5eca451bb2599f13f9f9197fd93b2a8bc2/datafusion/functions-array/src/extract.rs#L255),  due to "index out of bounds".
https://github.com/apache/datafusion/blob/ece7ae5eca451bb2599f13f9f9197fd93b2a8bc2/datafusion/functions-array/src/extract.rs#L253-L256

The fix is to use an empty string as the arg name if it is out of bounds,  allow `display_name` to pass through, and subsequently generating an error by checking the signature.
## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

**Review by jayzhan211 — APPROVED** [review]
👍

File: datafusion/sqllogictest/test_files/array.slt [hunk 1]
```diff
@@ -1974,6 +1978,10 @@ select array_slice(a, -1, 2, 1), array_slice(a, -1, 2),
 [] [] [] []
 [6.0] [6.0] [] []
 
+# Testing with empty arguments should result in an error
+query error DataFusion error: Error during planning: Error during planning: \[data_types_with_scalar_udf\] signature VariadicAny does not support zero arguments.
```
Comment (jayzhan211) [inline datafusion/sqllogictest/test_files/array.slt:1982]: It is handled by signature!

File: datafusion/functions-array/src/extract.rs [hunk 2]
```diff
@@ -97,8 +97,11 @@ impl ScalarUDFImpl for ArrayElement {
     }
 
     fn display_name(&self, args: &[Expr]) -> Result<String> {
-        let args_name: Vec<String> = args.iter().map(|e| e.to_string()).collect();
-        Ok(format!("{}[{}]", args_name[0], args_name[1]))
+        Ok(format!(
```
Comment (alamb) [inline datafusion/functions-array/src/extract.rs:100]: that is a nice drive by cleanup

**Comment by alamb** [comment]
Thanks @jonahgao and @jayzhan211

**Comment by jonahgao** [comment]
Thank you for the review @jayzhan211 @alamb

**Pull request merged**